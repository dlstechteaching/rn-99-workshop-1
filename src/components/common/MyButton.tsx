import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

interface MyButtonProps {
    label: string;
    onPress: () => void;
}

export const MyButton: React.VFC<MyButtonProps> = ({ label, onPress}) => {
    return (
            <TouchableOpacity onPress={onPress} style={styles.button}>
                <Text style={styles.text}>{ label }</Text>
            </TouchableOpacity>
    ) 
};

const styles = StyleSheet.create({
    button: {
        borderRadius: 3,
        padding: 10,
        backgroundColor: 'rgba(92, 45, 145, 1)',
        width: '100%',
        borderBottomEndRadius: 15,
        marginVertical: 20
      },
    text: {
        color: 'white',
        textAlign: 'center'
    }
});