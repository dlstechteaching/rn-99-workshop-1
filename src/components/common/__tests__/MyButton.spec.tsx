import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { MyButton } from '../MyButton';

describe('MyButton tests', () => {
	let component: ShallowWrapper;

	const onPressMock = jest.fn();
	it('should render properly', () => {
		givenComponent();
		thenItRendersProperly();
	});

	it('should call onPressMock', () => {
		givenComponent();
		whenButtonIsPressed();
		thenOnPressHasBeenCalled();
	});

	function givenComponent() {
		component = shallow(
			<MyButton label={'label'} onPress={onPressMock} />
		);
	}

	function whenButtonIsPressed() {
		component.find(TouchableOpacity).simulate('press');
	}

	function thenItRendersProperly() {
		expect(component).toMatchSnapshot();
	}

	function thenOnPressHasBeenCalled() {
		expect(onPressMock).toHaveBeenCalled();
	}
});