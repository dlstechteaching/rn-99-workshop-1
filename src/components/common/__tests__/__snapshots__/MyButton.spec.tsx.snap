// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton tests should render properly 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "rgba(92, 45, 145, 1)",
      "borderBottomEndRadius": 15,
      "borderRadius": 3,
      "marginVertical": 20,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "textAlign": "center",
      }
    }
  >
    label
  </Text>
</ForwardRef>
`;
